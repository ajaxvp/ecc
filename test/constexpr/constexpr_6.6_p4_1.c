/* ISO: 6.6 (4); value bound checks in constant expressions */

int a = 0x7FFFFFFF + 1;

int b = ((int) 0x80000000) - 1;

int c = 0x7FFFFFFF * 2;
