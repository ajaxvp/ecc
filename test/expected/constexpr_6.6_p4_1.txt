ecc: error: [3:23] in static initialization: addition in constant expression does not fit within its size
ecc: error: [5:31] in static initialization: subtraction in constant expression does not fit within its size
ecc: error: [7:23] in static initialization: multiplication in constant expression does not fit within its size
